Sources/NIOCore/BSDSocketAPI.swift

```diff
@@ -85,9 +87,15 @@ private let sysInet_pton: @convention(c) (CInt, UnsafePointer<CChar>?, UnsafeMut
 #endif
 
 #if os(Android)
+#if canImport(Android)
```

Comment:
Shouldn't we just use the same conditionals as above? Instead of checking for the os we just check for `canImport(Glibc)` and `canImport(Android)`?

Reply:
The OS check works when building for Android on both Swift 5 and 6 and also includes the Android armv7 definitions on line 100 below. It cannot be replaced because then `canImport(Glibc)` would be true and entered into on linux too, and the `#elseif os(Linux)` on line 103 would be skipped on most linux distros that use glibc.